platform/commonUI/edit/src/capabilities/TransactionalPersistenceCapability.js

```diff
@@ -67,10 +67,17 @@ define(
             }
 
             function onCancel() {
-                return self.persistenceCapability.refresh().then(function (result) {
+                if (self.domainObject.getModel().persisted) {
```

Comment:
In the interest of pedantic correctness, there should be a `!== undefined` on this check